valohai_yaml/objs/workload_resources.py

```diff
@@ -0,0 +1,111 @@
+from typing import Optional
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.types import SerializedDict
+
+
+class ResourceCPU(Item):
+    """CPU configuration."""
+
+    def __init__(
+        self,
+        max_value: Optional[int],
+        min_value: Optional[int],
+    ) -> None:
+        self.max = max_value
+        self.min = min_value
+
+    def __repr__(self) -> str:
+        """CPU data."""
+        return f'ResourceCPU("max": {self.max}, "min": {self.min})'
+
+
+class ResourceMemory(Item):
+    """Memory configuration."""
+
+    def __init__(
+        self,
+        max_value: Optional[int],
+        min_value: Optional[int],
+    ) -> None:
+        self.max = max_value
+        self.min = min_value
+
+    def __repr__(self) -> str:
+        """Memory data."""
+        return f'ResourceMemory("max": {self.max}, "min": {self.min})'
+
+
+class ResourceDevices(Item):
+    """Devices configuration."""
+
+    def __init__(self, devices: dict) -> None:
```

Comment:
not sure if `dict` is available, use `Dict` if not 🤔 

the project has `types.py` with various helpers that you can also use

Reply:
Changed to SerializedDict (which is what actually is passed to the function now).